src/scores.jl

```diff
@@ -472,11 +472,83 @@ function get_score_values(chain_data::ChainScoreData, score_name::String)
 end
 
 
+function district_and_plan_score_names(chain_data::ChainScoreData)::Array{String, 1}
+    """ Simple helper function for `save_scores`. Extracts the names of all of
+        the district/plan-level scores (including those nested within
+        CompositeScores).
+    """
+    score_names = String[]
+    for score in chain_data.scores
+        if score isa CompositeScore # add children score names
+            named_scores = filter(s -> isdefined(s, :name), score.scores)
+            append!(score_names, [s.name for s in named_scores])
+        else
+            push!(score_names, score.name)
+        end
+    end
+    return score_names
+end
+
+
 function save_scores(filename::String,
-                     chain_data::ChainScoreData)
-    """ Save the `scores` in a JSON file named `filename`.
+                     chain_data::ChainScoreData,
+                     score_names::Array{String,1}=String[])
+    """ Save the `scores` in a CSV file named `filename`. In order to avoid
```

Comment:
Docstrings should focus on the external API; this feels like an internal detail.

Reply:
i concur!

Reply:
edited!